tests/handlers/test_register.py

```diff
@@ -187,18 +187,30 @@ def test_auto_create_auto_join_rooms_when_support_user_exists(self):
 
     @defer.inlineCallbacks
     def test_auto_create_auto_join_where_no_consent(self):
-        """XXX what is this trying to test? I *think* it is trying to test
-        that we are not auto-joined to the server notices room if
-        block_events_without_consent_error is set, but (a) that doesn't seem to be
-        a sensible thing to test for, and (b) it doesn't work anyway because you can't
-        change the config after the EventCreationHandler has been instantiated.
+        """Test to ensure that the first user is not auto-joined to a room if
+        they have not given general consent.
         """
         self.hs.config.user_consent_at_registration = True
         self.hs.config.block_events_without_consent_error = "Error"
+
+        # Given:-
+        #    * a user must give consent,
+        #    * they have not given that consent
+        #    * The server is configured to auto-join to a room
+        # (and autocreate if necessary)
+        event_creation_handler = self.hs.get_event_creation_handler()
+        event_creation_handler._block_events_without_consent_error = (
```

Comment:
I'm a bit dubious about this poking into the guts of `EventCreationHandler` - it looks fragile. OTOH I can't really think of a better way of testing it, other than creating a new test class (possibly a subclass of `RegistrationTestCase`) which uses a different config - which might be a bit fiddly right now.

Reply:
Yeah this really mings - I'll comment explicitly to make it clearer